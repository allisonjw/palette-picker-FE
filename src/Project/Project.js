import React from 'react';
import { useDispatch } from 'react-redux';
import { deleteProject } from '../util/apiCalls';
import { deleteSelectedProject } from '../actions';
import './Project.scss';
import trashBin from '../images/rubbish.png';
import Palette from '../Palette/Palette';

const Project = ({name, id, palettes}) => {
    const dispatch = useDispatch();

    console.log('pro', id)

    // const remv = async (e) => {
    //   e.preventDefault();
    //   const newProject = { 
    //       project_name: newProjectName,  
    //   };
    //   addProject(newProject)
    //   dispatch(addNewProject(newProject));
    //   setNewProjectName('')

    // const removeProject = async () => {
    //   try {
    //     const newProjects = projects.filter(prj => prj.id !== project.id);
    //     setProjects(newProjects);
    //     await deleteProject(project.id);
    //   }
    //   catch (error) {
    //     console.log(error);
    //   }
    // };

    const removeProject = async () => {
      const body = {
        id: id
      }
      dispatch(deleteSelectedProject(id));
      deleteProject('projects', 'DELETE', body);
    };

    const displayPalettes = () => {
      return palettes.map(palette => {
        return <Palette {...palette} key={palette.id} />
      })
    } 
    return(
      <>
        <h3 className="project_name">{name}</h3>
            {displayPalettes()}
        <img
            className='deleteBtn'
            src={trashBin}
            alt='delete button'
            onClick={() => removeProject(id)}
        ></img>
      </>
    )
};

export default Project;