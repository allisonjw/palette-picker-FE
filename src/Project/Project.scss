.project_name {
  color: white;
  text-align: center;
  text-shadow: 1px 0px 7px #000000d6;
  display: inline-block;
  margin: 5px;
  font-family: 'Shadows Into Light', cursive;
}

.project-box {
  background-color: cadetblue;
  height: 148px;
  margin: 12px 10px 12px 10px;
  width: 300px;
  box-shadow: 8px 7px 12px 0px black;;
  border-radius: 2px;
}

.deleteBtn {
  height: 25px;
  margin: 2px;
  &:hover {
    cursor: pointer;
    transform: scale(1.1);
    transition: all .4s ease-in-out;
  }
}