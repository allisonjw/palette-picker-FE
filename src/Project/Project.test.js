import React from 'react';
import { render } from '@testing-library/react';
import Project from './Project';
import { shallow } from 'enzyme';

describe('Project', () => {
  let wrapper;
  const mockProjects =[{
        "id": 9,
        "project_name": "Jessie's House",
        "user_id": 5
    },
    {
        "id": 10,
        "project_name": "Dream Bath",
        "user_id": 5
    },
    {
        "id": 11,
        "project_name": "Allison's House",
        "user_id": 6    
  }];

  beforeEach(() => {
    wrapper = shallow(<Project projects={mockProjects}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});