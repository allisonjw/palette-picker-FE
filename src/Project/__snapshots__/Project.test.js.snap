// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot 1`] = `
<div>
  <section
    className="section_projects"
  >
    <h3
      className="project_name"
    >
      Jessie's House
      Dream Bath
      Allison's House
    </h3>
  </section>
  <img
    alt="delete button"
    className="deleteBtn"
    src="rubbish.png"
  />
  <Palette />
</div>
`;
